packages/skill-lesson/trpc/routers/lessons.ts

```diff
@@ -31,22 +31,27 @@ export const lessonsRouter = router({
 
       const lessons = section ? section.lessons : module.lessons
 
-      const exerciseForSolution = lessons.find((resource: SanityDocument) => {
-        return resource.solution?._key === lesson.solution?._key
-      })
-
-      // TODO: this is likely overloading the lesson type just because it
-      //   exists and is different that the exercise type. We should probably
-      //   create a new type instead of `lesson` since `lesson` describes ALL
-      //   lessons, not just the ones that are not exercises.
-      const current =
-        input.type === 'lesson'
-          ? find(lessons, {slug: input.slug})
-          : find(lessons, {_id: exerciseForSolution._id})
-      const nextExerciseIndex = indexOf(lessons, current) + 1
-
-      return lessons[nextExerciseIndex]
-        ? LessonResourceSchema.parse(lessons[nextExerciseIndex])
+      let currentLesson
+
+      switch (input.type) {
+        case 'exercise':
+          //exercises have solutions, that's what makes them exercises
+          //so we match the solution key to identify the exercise
+          const exerciseForSolution = lessons.find(
+            (resource: SanityDocument) => {
+              return resource.solution?._key === lesson.solution?._key
+            },
+          )
+          currentLesson = find(lessons, {_id: exerciseForSolution._id})
+          break
```

Comment:
an `exercise` is special because it has a `solution` and we need the `exercise` for a `solution` when that is the case. other types shouldn't require this so instead of looking for `lesson` and `explainer` we check for `exercise` and do the special business

this whole shit needs tests